### PR TITLE
Add limit parameter to fetch_*_table

### PR DESCRIPTION
**Pull request by AleksanderWWW**
## Before submitting checklist

- [x] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [ ] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

File: src/neptune/metadata_containers/metadata_container.py [hunk 1]
```diff
@@ -665,6 +667,8 @@ def _fetch_entries(self, child_type: ContainerType, query: NQLQuery, columns: Op
             columns=columns,
         )
 
+        leaderboard_entries = itertools.islice(leaderboard_entries, limit) if limit else leaderboard_entries
```
Comment (Raalsky) [inline src/neptune/metadata_containers/metadata_container.py:670]: I think that it's beneficial to still pass it to the pagination function as the limit may be much smaller than `step_size`. With the current approach, we're going to fetch for instance `100` entries even though the `limit` was set to `1`.

Comment (Raalsky) [reply src/neptune/metadata_containers/metadata_container.py:670]: You can probably just set `step_size` to a minimum of the `limit` and the old step_size. It should work as well.

File: src/neptune/internal/backends/hosted_neptune_backend.py [hunk 2]
```diff
@@ -1022,9 +1022,12 @@ def search_leaderboard_entries(
         types: Optional[Iterable[ContainerType]] = None,
         query: Optional[NQLQuery] = None,
         columns: Optional[Iterable[str]] = None,
+        limit: Optional[int] = None,
     ) -> Generator[LeaderboardEntry, None, None]:
         step_size = int(os.getenv(NEPTUNE_FETCH_TABLE_STEP_SIZE, "100"))
```
Comment (Raalsky) [inline src/neptune/internal/backends/hosted_neptune_backend.py:1027]: Let's rename it to `default_step_size` then.

**Comment by Raalsky** [comment]
@normandy7 we have a feature branch called `dev/fetch-ui-parity` that is going to have all the new parameters of `fetch_*_table` methods. What do you think about making just a one final docstrings review/adjustments after all of the parts will be merged?

**Comment by normandy7** [comment]
> @normandy7 we have a feature branch called `dev/fetch-ui-parity` that is going to have all the new parameters of `fetch_*_table` methods. What do you think about making just a one final docstrings review/adjustments after all of the parts will be merged?

@Raalsky Alright, that works. I'd like to understand the reasoning for "step size", as it's not entirely clear to me.

**Review by Raalsky — APPROVED** [review]

**Pull request merged**